### ISSUE TITLE
[Bug] xAxis sets category. When there are too many data points, thousands of data points will be displayed incompletely. （xAxis 设置 category，当数据点特别多时几千，前后数据展示不全））

### ISSUE THREAD

**Issue by FriendsA** [apache/echarts#19658]
### Version

5.5.0

### Link to Minimal Reproduction

https://codepen.io/FriendsA/pen/VwNZxmw

### Steps to Reproduce

1. 点击提供的用例链接
2. 或者使用此配置在官方的链接中覆盖config查看 https://echarts.apache.org/examples/en/editor.html?c=line-stack
[config.txt](https://github.com/apache/echarts/files/14430083/config.txt)

### Current Behavior

开始和结束部分的数据都会缺失

### Expected Behavior

希望都可以正常展示出来

### Environment

```markdown
- OS:macOS Big Sur、window
- Browser:Chrome122.0.6261.69
- Framework: React@17
```

**Comment by MatthiasMert** [comment]
Your data contains a large number of `null` values at start and end. The property [connectNulls](https://echarts.apache.org/en/option.html#series-line.connectNulls) interpolates the line if **inbetween** two datapoints there a re `null` values. In Your case there is no datapoint to connect to in the beginning. You could add an auxiliary point at the beginning to combat that. Additionally, when zooming in, all data points outside of the view are filtered out which leads to the same problem. You can add [filterMode](https://echarts.apache.org/en/option.html#dataZoom-slider.filterMode) `'none'` to not filter out data but this will slow down performance.

**Comment by FriendsA** [comment]
> Your data contains a large number of `null` values at start and end. The property [connectNulls](https://echarts.apache.org/en/option.html#series-line.connectNulls) interpolates the line if **inbetween** two datapoints there are `null` values. In Your case there is no datapoint to connect to in the beginning. You could add an auxiliary point at the beginning to combat that. Additionally, when zooming in, all data points outside of the view are filtered out which leads to the same problem. You can add [filterMode](https://echarts.apache.org/en/option.html#dataZoom-slider.filterMode) `'none'` to not filter out data but this will slow down performance.

thinks for your help! you can hidden the line which have `null` data and leave the other complete line only，it  also can not display the last point which the date at 2022-09-26.  I want it can display last point on chart!

**Comment by FriendsA** [comment]
> > it also can not display the last point which the date at 2022-09-26.
> 
> The last point is displayed. Only if you zoom into the area which contains `null` values the line is not shown which can be fixed with `filterMode: 'none'`.
> 
> Did you apply the suggestions i gave? 

yes ,i try.If I'm not mistaken i think What you mean is to add a configuration `filterMode: 'none'`， I fixed the example. https://codepen.io/FriendsA/pen/VwNZxmw
 do you mean like this? or can you fix this configuration file and reply to me？so I can understand it directly
![Uploading Snipaste_2024-02-28_16-29-14.png…]()

**Comment by MatthiasMert** [comment]
This looks good to me.

What issue do you still have?

**Comment by FriendsA** [comment]
> This looks good to me.
> 
> What issue do you still have?

it is still lost last point to me.
![Snipaste_2024-02-28_16-29-14](https://github.com/apache/echarts/assets/20056209/08470ea6-aa32-465b-9d03-96864d955b8c)

**Comment by MatthiasMert** [comment]
You just cant hover over it in the small view (since there are more data points than pixels). If you increase the width or zoom in you will see that it is there.

**Comment by FriendsA** [comment]
> You just cant hover over it in the small view (since there are more data points than pixels). If you increase the width or zoom in you will see that it is present.

yes，but just on both sides has，so i think it is a bug. anyway, your help is greatly appreciated！

**Comment by MatthiasMert** [comment]
I dont understand, sry.

**Comment by FriendsA** [comment]
> yes，but just on both sides has，


what I mean is that this phenomenon only appears at both ends of the chart. It should be a bug.

**Comment by FriendsA** [comment]
> I dont understand, sry.

what I mean is that this phenomenon only appears at both ends of the chart. It should be a bug.

**Comment by MatthiasMert** [comment]
It does not only appear on both ends. If you hover through the data the tooltip will skip a lot of points

**Comment by miracleren** [comment]
#main {
  width: 1500px;
  height: 400px;
  border: 1px solid red;
}

因为描点太多，宽度像素不够，会导致tooltip跳过某些点导致显示不出来，把宽度调宽点就可以了。
